NEW_CONNECTION.md

```diff
@@ -0,0 +1,102 @@
+# Add New Data Source
+
+The following are the instructions for adding a new data source 
+
+## The following are the files that need to be updated
+
+1. app/constants/constants.js
+2. app/actions/sessions.js
+3. app/components/Settings/Tabs/Tab.react.js
+4. app/components/Settings/Settings.react.js
+5. backend/persistent/datastores/Datastores.js
+
+## The following are the new files that need to be added
+
+1. app/images
+2. backend/persistent/datastores/NameOfDatastoreToConnect.js
+
+# Detailed Implementation Instructions
+
+## Images
+It is recommended to add the images as png and a good idea to include a regular and small icon.
+You can add the new icon in the directory app/images.  It recommended to use png format
+and name the file in lower case with dashes (ex: mysql-logo.png )
+
+## Updating the constants.js file
+The following are the instructions for adding the Constants File:
+1. Add an entry in DIALECTS
+2. Add an entry in SQL_DIALECTS_USING_EDITOR
+3. Add an entry under CONNECTION_CONFIG
+4. Add the location of the logo in 
+5. Add the preview query in the PREVIEW_QUERY section
+6. Add an entry in the SAMPLE_DBS 
+
+## Updating the Tab.react.js 
+The following are the instructions for adding the Tab React JS file
+1. Add an entry for the connection label
+
+## Updating the Settings.react.js
+Update the fetchData function to make the calls to appropriate sessions.js backend.  
+Currently there is Dialect types for SQL, ElasticSearch, S3 and Apache Drill.  If 
+your new data source is an SQL Type you will need to add the new Dialect in the 
+fetchData section for SQL types.  
+
```

Comment:
I should clean up this file.

For the time being, I'd keep things simple and just ask the developer to update `fetchData` and add their connector dialect to the list in:

```js
        if (contains(connectionObject.dialect, [
                    DIALECTS.APACHE_IMPALA,
                    DIALECTS.APACHE_SPARK,
                    DIALECTS.IBM_DB2,
                    DIALECTS.MYSQL, DIALECTS.MARIADB, DIALECTS.POSTGRES,
                    DIALECTS.REDSHIFT, DIALECTS.MSSQL, DIALECTS.SQLITE,
                    DIALECTS.DATA_WORLD
        ])) {
```